plugins/woocommerce-admin/client/core-profiler/pages/BusinessInfo.tsx

```diff
@@ -1,34 +1,340 @@
+				<Heading
+					className="woocommerce-profiler__stepper-heading"
+					title={ __(
+						'Tell us a bit about your store',
+						'woocommerce'
+					) }
+					subTitle={ __(
+						"We'll use this information to help you set up payments, shipping, and taxes, as well as recommending the best theme for your store.",
+						'woocommerce'
+					) }
+				/>
+
+				<form
+					className="woocommerce-profiler-business-information-form"
+					autoComplete="off"
+				>
+					<TextControl
+						className="woocommerce-profiler-business-info-store-name"
+						onChange={ ( value ) => {
+							setStoreName( value );
+						} }
+						value={ storeName }
+						label={
+							<>
+								{ __(
+									'Give your store a name',
+									'woocommerce'
+								) }
+							</>
+						}
+						placeholder={ __(
+							'Ex. My awesome store',
+							'woocommerce'
+						) }
+					/>
+					<p className="woocommerce-profiler-question-subtext">
+						{ __(
+							"Don't worry — you can always change it later!",
+							'woocommerce'
+						) }
+					</p>
+					<p className="woocommerce-profiler-question-label">
+						{ selectIndustryQuestionLabel }
+					</p>
+					<SelectControl
+						className="woocommerce-profiler-select-control__industry"
+						instanceId={ 1 }
+						placeholder={ __(
+							'Select an industry',
+							'woocommerce'
+						) }
+						label={ __( 'Select an industry', 'woocommerce' ) }
+						options={ industryChoices }
+						excludeSelectedOptions={ false }
+						help={ <Icon icon={ chevronDown } /> }
+						onChange={ (
+							results: Array< typeof industryChoices[ number ] >
+						) => {
+							if ( results.length ) {
+								setIndustry( results[ 0 ] );
+							}
+						} }
+						selected={ industry ? [ industry ] : [] }
+						showAllOnFocus
+						isSearchable
+					/>
+					<p className="woocommerce-profiler-question-label">
+						{ __( 'Where is your store located?', 'woocommerce' ) }
+						<span className="woocommerce-profiler-question-required">
+							{ '*' }
+						</span>
+					</p>
+					<SelectControl
+						className="woocommerce-profiler-select-control__country"
+						instanceId={ 2 }
+						placeholder={ selectCountryLabel }
+						label={
+							storeCountry.key === '' ? selectCountryLabel : ''
+						}
+						getSearchExpression={ ( query: string ) => {
+							return new RegExp(
+								'(^' + query + '| — (' + query + '))',
+								'i'
+							);
+						} }
+						options={ countries }
+						excludeSelectedOptions={ false }
+						help={ <Icon icon={ chevronDown } /> }
+						onChange={ ( results: Array< CountryStateOption > ) => {
+							if ( results.length ) {
+								setStoreCountry( results[ 0 ] );
+							}
+						} }
+						selected={ storeCountry ? [ storeCountry ] : [] }
+						showAllOnFocus
+						isSearchable
+					/>
+					{ geolocationOverruled && ! dismissedGeolocationNotice && (
```

Comment:
I'm not sure why. After switching back to the original country, the notice still appears.

![Screenshot 2023-05-25 at 18 22 55](https://github.com/woocommerce/woocommerce/assets/4344253/d60d7eaa-13a1-4dbe-a4dd-f3b5ef7650e9)

Reply:
Thanks, great catch! That was a pretty major bug, turns out geolocationMatch was always an empty string if the country was restored from options...

Fixed in https://github.com/woocommerce/woocommerce/pull/38412/commits/4c3e694332bde603017409c84ae2966f1785aa94